File: vscode/src/tree-sitter/query-tests/test-data/intents.py
def wrapper():
    print('wrapper')
    def test():
        #     |
        pass

# ------------------------------------

def test_params():
    #          |
    wrapper()

# ------------------------------------

def test_parameter(val):
    #              |
    wrapper()

# ------------------------------------

def arrow_wrapper():
    arrow = lambda value: None
    #                   |

# ------------------------------------

class Agent:
    #      |
    pass

# ------------------------------------

def signature():
    #          |
    pass

# ------------------------------------

# comment
# |

# ------------------------------------

def function_name():
    #           |
    pass

# ------------------------------------

def return_statement():
    return
    #    |

# ------------------------------------

def return_statement_value(value: str, flag: bool = False):
    return 'asd'
    #      |

# ------------------------------------

return_statement_value('value')
#                      |

# ------------------------------------

return_statement_value('value', False)
#                               |

# ------------------------------------

return_statement_value()
#                     |

# ------------------------------------

from math import cos
#    |

# ------------------------------------

from math import comb
#                |

File: vscode/src/tree-sitter/query-tests/test-data/intents.ts
function wrapper() {
    console.log('wrapper')
    function test() {
        //          |
    }
}

// ------------------------------------

function testParams() {
    //             |
    wrapper
}

// ------------------------------------

function testParameter(val) {
    //                 |
    wrapper
}

// ------------------------------------

function arrowWrapper() {
    const arrow = (value: string) => {
        //                           |
    }
}

// ------------------------------------

class Agent {
    //      |
}

// ------------------------------------

function signature()
//                |

// ------------------------------------

// comment
//       |

// ------------------------------------

/**
 * comment
 //      |
 */

// ------------------------------------

function functionName() {}
//                  |

// ------------------------------------

function returnStatement() {
    return
    //   |
}

// ------------------------------------

function returnStatementValue(value: string, flag?: boolean) {
    return "asd"
    //     |
}

// ------------------------------------

interface TestInterface {
    //                  |
}

// ------------------------------------

type TestType = {
    //          |
}

// ------------------------------------

returnStatementValue("value")
//                   |

// ------------------------------------

returnStatementValue("value", false)
//                            |

// ------------------------------------

returnStatementValue()
//                  |

// ------------------------------------

const object = {
    key: "value"
    //   |
}

// ------------------------------------

returnStatementValue("value", () => {
    //                              |
    const value = "value"
})

// ------------------------------------

returnStatementValue("value", {key: value})
//                                  |

// ------------------------------------

returnStatementValue("value", () => {
    const value = "value"
   //             |
})
